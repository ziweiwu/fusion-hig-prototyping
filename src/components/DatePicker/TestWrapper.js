import React from "react";
import PropTypes from "prop-types";

const TestWrapper = ({ children }) => (
  <div className="test-wrapper">{children}</div>
);

TestWrapper.propTypes = {
  children: PropTypes.node
};

export default TestWrapper;
