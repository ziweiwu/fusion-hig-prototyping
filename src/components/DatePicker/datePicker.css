@import url(~@hig/styles/build/index.css);
/* Dimensions */
/* Colors */
.hig__text-field__label {
  position: static; }

.react-datepicker__day--outside-month {
  visibility: hidden; }

.react-datepicker__header {
  background-color: white; }

.react-datepicker__day:hover {
  background-color: #9BD5EF; }

.react-datepicker__day--keyboard-selected {
  background-color: #0696d7; }

.react-datepicker__day {
  font-size: 0.8rem; }

.react-datepicker__current-month,
.react-datepicker-time__header {
  font-size: 1.2rem; }

/* Clear button icon */
button.react-datepicker__close-icon::after {
  margin: -10px auto 0; }

/* Hig TextField Style */
hig__text-field__input-wrapper {
  margin-bottom: 0px; }

.hig__text-field {
  min-width: 240px;
  max-width: 240px; }

.hig__text-field__input {
  padding-bottom: 2px; }

.hig__text-field__label {
  padding-bottom: 8px; }

.hig__text-field__clear-button {
  top: 26px; }
