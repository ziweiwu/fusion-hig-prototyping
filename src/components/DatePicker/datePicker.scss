@import "~@hig/styles/build/index.css";

/* Dimensions */
$datepicker__border-radius: 0.2rem;
$datepicker__day-margin: 0.166rem;
$datepicker__font-family: ArtifaktElement, sans-serif;
$datepicker__font-size: 0.8rem;
$datepicker__item-size: 1.7rem;
$datepicker__margin: 0.4rem;
$datepicker__navigation-size: 0.45rem;
$datepicker__triangle-size: 8px;

/* Colors */
$datepicker__background-color: white;
$datepicker__border-color: #D4DBE1;
$datepicker__highlighted-color: #87BC40;
$datepicker__muted-color: #999999;
$datepicker__outsideMonth-color: #999999;
$datepicker__selected-color: #0696d7;
$datepicker__text-color: #000;
$datepicker__header-color: #000;
$datepicker__navigation-disabled-color: lighten($datepicker__muted-color, 20%);

// hide the days from prev and next month
.react-datepicker__day--outside-month {
  visibility: hidden;
}

.react-datepicker__day {
  width: 2.5em;
  border-radius: 0 !important;
  margin: 0 !important;
  width: 30px !important;
  height: 30px !important;

  // strong emphasis on today's date with font weight
  &--today {
    font-weight: bold !important;
  }
  // hover color for calender day
  &:hover {
    background-color: lighten($datepicker__selected-color, 20%) !important;
  }

  &--disabled {
    border-right-color: $datepicker__navigation-disabled-color;

    // disable hover background color for disable state
    &:hover {
      background-color: white !important;
    }
  }

  // color for dates in range selection
  &--in-selecting-range:not(&--in--range) {
    background-color: lighten($datepicker__selected-color, 30%) !important;
  }
}

// font size for portal view header
.react-datepicker__current-month,
.react-datepicker-time__header {
  font-size: $datepicker__font-size * 1.5;
}

/* Clear button icon */
button.react-datepicker__close-icon::after {
  margin: -10px auto 0;
}

// using hig chevron right and left icon for navigation
// used svg uri converter to convert svg to background images
button.react-datepicker__navigation {
  line-height: 1.7rem;
  text-align: center;
  cursor: pointer;
  position: absolute;
  //top: 5px;
  //width: 25px !important;
  //height: 25px !important;
  top: 10px;
  width: 15px !important;
  height: 15px !important;
  padding: 0;
  border: none !important;
  z-index: 1;

  //&--previous {
  //  background: url("data:image/svg+xml;charset=utf8,%3Csvg id='assets' xmlns='http://www.w3.org/2000/svg' width='24' height='24' viewBox='0 0 24 24'%3E%3Ctitle%3Echevron-left-24%3C/title%3E%3Cpath d='M11.17,12l5.66-5.66a1,1,0,0,0,0-1.41L14.71,2.81a1,1,0,0,0-1.41,0L4.81,11.29a1,1,0,0,0,0,1.41l8.49,8.49a1,1,0,0,0,1.41,0l2.12-2.12a1,1,0,0,0,0-1.41ZM14,20.49,5.51,12,14,3.51l2.12,2.12L9.76,12l6.36,6.36Z'/%3E%3C/svg%3E") !important;
  //}
  //&--next {
  //  background-image: url("data:image/svg+xml;charset=utf8,%3Csvg id='assets' xmlns='http://www.w3.org/2000/svg' width='24' height='24' viewBox='0 0 24 24'%3E%3Ctitle%3Echevron-right-24%3C/title%3E%3Cpath d='M19.19,11.29,10.71,2.81a1,1,0,0,0-1.41,0L7.17,4.93a1,1,0,0,0,0,1.41L12.83,12,7.17,17.66a1,1,0,0,0,0,1.41l2.12,2.12a1,1,0,0,0,1.41,0l8.49-8.49A1,1,0,0,0,19.19,11.29ZM10,20.49,7.88,18.36,14.24,12,7.88,5.64,10,3.51,18.49,12Z'/%3E%3C/svg%3E") !important;
  //}

  &--previous {
    background-image: url("data:image/svg+xml,%3Csvg id='assets' xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Ctitle%3Ecaret-left-16%3C/title%3E%3Cpolygon points='3.65 8.09 9.95 1.79 11.72 3.56 7.18 8.09 11.72 12.63 9.95 14.4 3.65 8.09'/%3E%3C/svg%3E");
  }
  &--next {
    background-image: url("data:image/svg+xml,%3Csvg id='assets' xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Ctitle%3Ecaret-right-16%3C/title%3E%3Cpolygon points='4.28 12.63 8.82 8.09 4.28 3.56 6.05 1.79 12.35 8.09 6.05 14.4 4.28 12.63'/%3E%3C/svg%3E");
  }
}

/* Hig TextField Style */
// static set the position of label text of input to remain on top.
// reflective set the position of label text to be dynamically positioned.
.hig__text-field__label-v1 {
  position: static;
}

//calender icon
.hig__text-field__icon-v1 {
  bottom: 3px
}

.hig__icon-button {
  background-color: transparent;
  border: none;
  padding-top: 5px;
}

.hig__text-field__clear {
  top: 15px;
  right: 5px;
}

.hig__text-field{
  min-width: 230px;
  max-width: 230px
}







/***********************************************
//turn off focus animation of text field
***********************************************/
//.hig__text-field-v1__input {
//  display: block;
//  padding: 0;
//  width: 100%;
//  box-sizing: border-box;
//  background-color: transparent;
//  border: none;
//
//  &:focus {
//    outline: none;
//    @include placeholder {
//      color: color(hig-cool-gray-50);
//      font-size: 14px;
//      transition-delay: 5s;
//    }
//  }
//}
//
//.hig__text-field-v1__label {
//  @include typography-medium;
//  position: absolute;
//  transition-property: none !important;
//  transition-duration: 0s, 0s, 0s;
//  transition-delay: 5s;
//
//  .hig__text-field-v1--disabled & {
//    color: color(hig-cool-gray-20);
//    pointer-events: none;
//  }
//}
//
//.hig__text-field-v1__label--input-focused,
//.hig__text-field-v1__label--with-value {
//  @include typography-medium;
//  color: color(hig-blue-60);
//  font-size: 14px;
//  top: 0;
//}
//

//.hig__text-field-v1__input-wrapper {
//  @include styled-borer(color(hig-blue-60));
//}

//hig__text-field__input-wrapper {
//  margin-bottom: 0px;
//}
//
//.hig__text-field__input {
//  padding-bottom: 2px;
//}
//
//.hig__text-field__label {
//  padding-bottom: 8px;
//}

// The above code will override the stylesheet from react-datepicker library
@import 'node_modules/react-datepicker/src/stylesheets/datepicker';

