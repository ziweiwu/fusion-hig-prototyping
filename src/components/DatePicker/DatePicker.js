import React from 'react';
import PropTypes from "prop-types";
import '@hig/styles/build/index.css';
import '@hig/text-field/build/index.css';
import {TextFieldPresenter} from '@hig/text-field';
import ReactDatePicker from 'react-datepicker';
import './datePicker.css';

export default class DatePicker extends React.Component {
  static propTypes = {
    adjustDateOnChange: PropTypes.bool,
    allowSameDay: PropTypes.bool,
    autoComplete: PropTypes.string,
    autoFocus: PropTypes.bool,
    calendarClassName: PropTypes.string,
    calendarContainer: PropTypes.func,
    children: PropTypes.node,
    className: PropTypes.string,
    customInput: PropTypes.element,
    customInputRef: PropTypes.string,
    dateFormat: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    dateFormatCalendar: PropTypes.string,
    dayClassName: PropTypes.func,
    disabled: PropTypes.bool,
    endDate: PropTypes.object,
    excludeDates: PropTypes.array,
    filterDate: PropTypes.func,
    fixedHeight: PropTypes.bool,
    formatWeekNumber: PropTypes.func,
    highlightDates: PropTypes.array,
    id: PropTypes.string,
    includeDates: PropTypes.array,
    isClearable: PropTypes.bool,
    locale: PropTypes.string,
    maxDate: PropTypes.object,
    minDate: PropTypes.object,
    monthsShown: PropTypes.number,
    name: PropTypes.string,
    onChange: PropTypes.func,
    onSelect: PropTypes.func,
    onWeekSelect: PropTypes.func,
    openToDate: PropTypes.object,
    placeholderText: PropTypes.string,
    readOnly: PropTypes.bool,
    required: PropTypes.bool,
    selected: PropTypes.object,
    selectsEnd: PropTypes.bool,
    selectsStart: PropTypes.bool,
    showWeekNumbers: PropTypes.bool,
    startDate: PropTypes.object,
    startOpen: PropTypes.bool,
    title: PropTypes.string,
    todayButton: PropTypes.string,
    useWeekdaysShort: PropTypes.bool,
    formatWeekDay: PropTypes.func,
    utcOffset: PropTypes.number,
    value: PropTypes.string,
    weekLabel: PropTypes.string,
    shouldCloseOnSelect: PropTypes.bool,
    minTime: PropTypes.object,
    maxTime: PropTypes.object,
    readOnly: PropTypes.bool
  };

  render() {
    const props = this.props;

    return (<ReactDatePicker
      {...props}
      readOnly
      fixedHeight
      showMonthYearDropdown={false}
      showMonthDropdown={false}
      showYearDropdown={false}
      showTimeSelect={false}
      isClearable={false}

      // use TextField as inputField
      // use ref to allow the use clear button in TextField component
      // instead of the one comes with ReactDatePicker
      ref={node => this.node = node}
      customInput={<TextFieldPresenter
        {...props}
        id={props.selected.toString()} //use selected id to keep input id unique
        showClearButton={props.isClearable}
        onClearButtonClick={() => {
          this.node.clear();
        }}
      />}

      // calender popper settings
      popperPlacement="bottom-start"
      popperModifiers={{
        // adjust position of calender popper, (horizontal, vertical)
        offset: {
          enabled: true,
          offset: '0px, -30px',
        },
        flip: {
          enabled: false,
        },
        preventOverflow: {
          enabled: true,
          escapeWithReference: false,
        },
      }}
    />);
  }
}

