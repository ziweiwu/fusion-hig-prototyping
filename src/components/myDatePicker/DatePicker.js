import React from 'react';
import ReactDatePicker from 'react-datepicker';
import TextField from '@hig/text-field';
import '@hig/text-field/build/index.css';
import './stylesheets/datePicker.css';

const DatePicker = (props) => {
  return <ReactDatePicker
    {...props}
    fixedHeight={true}
    customInput={<TextField label={props.label} placeholderText={props.placeholderText}/>}
    popperPlacement="bottom-start"

    popperModifiers={{
      //adjust position of calender popper, (horizontal, vertical)
      offset: {
        enabled: true,
        offset: '0px, -30px'
      },

      //prevent calender popper to flip
      flip: {
        enabled: false
      },

      preventOverflow: {
        enabled: true,
        escapeWithReference: false
      }
    }}


  />
};

export default DatePicker;