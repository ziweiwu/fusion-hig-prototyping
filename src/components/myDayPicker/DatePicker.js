import React from 'react';
import ReactDatePicker from 'react-datepicker';
import TextField from '@hig/text-field';
import '@hig/text-field/build/index.css';
import './styles/datePicker.css';

const DatePicker = (props) => {
  return <ReactDatePicker {...props}   customInput={<TextField />}/>
};

export default DatePicker;