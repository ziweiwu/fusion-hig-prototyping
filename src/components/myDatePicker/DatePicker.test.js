import React from "react";
import {mount, shallow} from "enzyme";
import moment from "moment";
import DatePicker from "./index";
import propagateToGlobal from "./test_setup"
import sinon from 'sinon';

propagateToGlobal();

const fn = jest.fn();

describe("DatePicker", () => {
  //snapshot testing
  test('Snapshot tests regular view', () => {
    const wrapper = shallow(<DatePicker selected={moment("2017-09-15 09:30:00")}/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot tests portal view', () => {
    const wrapper = shallow(<DatePicker selected={moment("2017-09-15 09:30:00")} withPortal/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot tests disabled view', () => {
    const wrapper = shallow(<DatePicker selected={moment("2017-09-15 09:30:00")} disabled/>);
    expect(wrapper).toMatchSnapshot();
  });

  //component render testing
  it("Render the dayPicker component correctly.", () => {
    const wrapper = shallow(<DatePicker/>);
    expect(wrapper).toHaveLength(1);
  });

  it("Render the dayPicker input field correctly.", () => {
    const wrapper = mount(<DatePicker/>);
    expect(wrapper.find("TextField").length).toBe(1);
  });

  it("Render the label correctly.", () => {
    const wrapper = mount(<DatePicker label="foo"/>);
    expect(wrapper.instance().props.label).toBe("foo");
  });

  it("Render the placeholder correctly.", () => {
    const wrapper = mount(<DatePicker placeholderText="bar"/>);
    expect(wrapper.instance().props.placeholderText).toBe("bar");
  });


  //test interactions
  it("Show the calendar when focusing on the date input", () => {
    const wrapper = mount(<DatePicker/>);
    const input = wrapper.find("TextField");
    input.props().onFocus();
    expect(wrapper.find(".react-datepicker-popper").length).toBe(1);
  });

  it("Disable calender popper when disable is selected", () => {
    const wrapper = mount(<DatePicker disabled/>);
    const input = wrapper.find("TextField");
    input.props().onFocus();
    expect(wrapper.find(".react-datepicker-popper").length).toBe(0);
  });

  // it("Clear input field when clear button is clicked ", () => {
  //   const wrapper = mount(<DatePicker isClearable/>);
  //   const clearButton = wrapper.find("hig__text-field__clear-button");
  //   clearButton.props().onClick();
  //   expect(wrapper.find.props.value).toBe(null);
  // });
});



