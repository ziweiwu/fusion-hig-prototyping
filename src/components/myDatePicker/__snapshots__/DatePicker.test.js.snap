// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Snapshot tests disabled view 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DatePicker
    allowSameDay={false}
    customInput={
        <TextField
            label={undefined}
            placeholderText={undefined}
        />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={true}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    fixedHeight={true}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
        Object {
            "flip": Object {
              "enabled": false,
            },
            "offset": Object {
              "enabled": true,
              "offset": "0px, -30px",
            },
            "preventOverflow": Object {
              "enabled": true,
              "escapeWithReference": false,
            },
          }
    }
    popperPlacement="bottom-start"
    preventOpenOnFocus={false}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
/>,
  "nodes": Array [
    <DatePicker
      allowSameDay={false}
      customInput={
            <TextField
                  label={undefined}
                  placeholderText={undefined}
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={true}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, -30px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showTimeSelect={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DatePicker
      disabled={true}
      selected={"2017-09-15T13:30:00.000Z"}
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <DatePicker
      allowSameDay={false}
      customInput={
            <TextField
                  label={undefined}
                  placeholderText={undefined}
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={true}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, -30px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showTimeSelect={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DatePicker
    disabled={true}
    selected={"2017-09-15T13:30:00.000Z"}
/>,
}
`;

exports[`DatePicker Snapshot tests portal view 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DatePicker
    allowSameDay={false}
    customInput={
        <TextField
            label={undefined}
            placeholderText={undefined}
        />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    fixedHeight={true}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
        Object {
            "flip": Object {
              "enabled": false,
            },
            "offset": Object {
              "enabled": true,
              "offset": "0px, -30px",
            },
            "preventOverflow": Object {
              "enabled": true,
              "escapeWithReference": false,
            },
          }
    }
    popperPlacement="bottom-start"
    preventOpenOnFocus={false}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={true}
/>,
  "nodes": Array [
    <DatePicker
      allowSameDay={false}
      customInput={
            <TextField
                  label={undefined}
                  placeholderText={undefined}
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, -30px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showTimeSelect={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={true}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DatePicker
      selected={"2017-09-15T13:30:00.000Z"}
      withPortal={true}
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <DatePicker
      allowSameDay={false}
      customInput={
            <TextField
                  label={undefined}
                  placeholderText={undefined}
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, -30px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showTimeSelect={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={true}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DatePicker
    selected={"2017-09-15T13:30:00.000Z"}
    withPortal={true}
/>,
}
`;

exports[`DatePicker Snapshot tests regular view 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DatePicker
    allowSameDay={false}
    customInput={
        <TextField
            label={undefined}
            placeholderText={undefined}
        />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    fixedHeight={true}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
        Object {
            "flip": Object {
              "enabled": false,
            },
            "offset": Object {
              "enabled": true,
              "offset": "0px, -30px",
            },
            "preventOverflow": Object {
              "enabled": true,
              "escapeWithReference": false,
            },
          }
    }
    popperPlacement="bottom-start"
    preventOpenOnFocus={false}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
/>,
  "nodes": Array [
    <DatePicker
      allowSameDay={false}
      customInput={
            <TextField
                  label={undefined}
                  placeholderText={undefined}
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, -30px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showTimeSelect={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DatePicker
      selected={"2017-09-15T13:30:00.000Z"}
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <DatePicker
      allowSameDay={false}
      customInput={
            <TextField
                  label={undefined}
                  placeholderText={undefined}
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, -30px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showTimeSelect={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DatePicker
    selected={"2017-09-15T13:30:00.000Z"}
/>,
}
`;
